src/Core/System/SalesChannel/Context/BaseSalesChannelContextFactory.php

```diff
@@ -73,10 +76,6 @@ public function create(string $salesChannelId, array $options = []): BaseSalesCh
         $criteria->setTitle('base-context-factory::sales-channel');
         $criteria->addAssociation('currency');
         $criteria->addAssociation('domains');
-        $criteria->getAssociation('languages')
-            ->addFilter(new EqualsFilter('id', $context->getLanguageId()))
-            ->addAssociation('translationCode')
-            ->addAssociation('locale');
```

Comment:
People have relied on `$context->getSalesChannel()->getLanguages()` to have things. My view: this is a BC and needs to be put behind `v6.8.0.0`

Reply:
yeah, I need to rethink the whole stuff again. I also discussed several things with Olli. I should write them down in a ticket already :sweat_smile: